src/Service.Tests/SqlTests/GraphQLMutationTests/MultipleMutationIntegrationTests.cs

```diff
@@ -0,0 +1,327 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Azure.DataApiBuilder.Service.Tests.SqlTests;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Tests.Authorization.GraphQL
+{
+    [TestClass, TestCategory(TestCategory.MSSQL)]
+    public class MultipleMutationIntegrationTests : SqlTestBase
+    {
+        /// <summary>
+        /// Set the database engine for the tests
+        /// </summary>
+        [ClassInitialize]
+        public static async Task SetupAsync(TestContext context)
+        {
+            DatabaseEngine = TestCategory.MSSQL;
+            await InitializeTestFixture();
+        }
+
+        /// <summary>
+        /// Test to validate that for 'create one' mutations, we throw an appropriate exception when the user request provides
+        /// no source of value for a non-nullable and non-default referencing column.
+        /// There are 3 sources of values for a referencing column:
+        /// 1. Explicit value assignment to the referencing column,
+        /// 2. Value derived from the parent referenced entity,
+        /// 3. Providing value for inserting record in the referenced target entity, in which case value for referencing column
+        /// is derived from the value of corresponding referenced column after insertion in the referenced entity.
+        /// When the user provides none of the above, we throw an exception.
+        /// </summary>
+        [TestMethod]
+        public async Task AbsenceOfSourceOfTruthForReferencingColumnForCreateOneMutations()
+        {
+            // For a relationship between Book (Referencing) - Publisher (Referenced) defined as books.publisher_id -> publisher.id,
+            // in the request input:
+            // 1. No explicit value is provided for referencing field publisher_id
+            // 2. Book is the top-level entity - so there is no parent entity.
+            // 3. No value is assigned for the referenced entity via relationship field ('publisher').
+            string createOneBookMutationName = "createbook";
+            string multipleCreateOneBookWithoutPublisher = @"mutation {
+                    createbook(item: { title: ""My New Book"" }) {
```

Comment:
if book.publisher_id were nullable... is that possible / what would happen? would multiple insert validation still reject the request?

Reply:
No, it would not reject. If there are no sources of value for a referencing field, it means you are not performing a related insertion. You are just performing an insertion on the independent Book entity. And it is fine to insert null value for publisher_id then - because it is not used to relate the records with records inserted in publisher entity.